PyTorch migration: Finishing touches

timokau opened the pull request:
## Description

This collects some misc fixes, changes and updates regarding the pytorch migration. I have updated the status in the README. The pytorch version does not have feature parity, but it is now at a point where it should be a viable replacement for at least some of the use cases.

## Motivation and Context

Part of the ongoing pytorch migration.

## How Has This Been Tested?

I ran the test suite and used our set of linters.

## Does this close/impact existing issues?

Related to #125. I think the issue can be closed once the pytorch migration branch has been merged into master.

## Types of changes

This contains various documentation changes as well as a breaking change to the CmpNet estimators.

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

kiudee commented on csrank/choicefunction/fate_choice.py:
```diff
@@ -10,6 +10,8 @@
 class FATEChoiceFunction(SkorchChoiceFunction):
     """A variable choice estimator based on the FATE-Approach.
 
+    See the documentation of the ``FATEScoring`` class for more details.
```
It could potentially be possible to cross-reference the FATEScoring class:
https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#cross-referencing-syntax

But I am not sure how well that works.

kiudee commented on HISTORY.rst:
```diff
@@ -14,8 +14,8 @@ History
 * Behavior and default parameters of the estimators may differ from the
-  previous versions. For example the default RankNet activation is now SELU
-  instead of ReLU.
+  previous versions. For example the default activation for CMpNet and RankNet
```
"CMpNet" -> "CmpNet"

timokau replied:
Fixed.

kiudee reviewed (changes requested):
Just found a minor typo. The cross-referencing suggestion is not that important.

timokau commented:
I fixed the typo, added cross-referencing, fixed the `choicefunctions` module name in the api docs and added the missing estimators to the api docs.

The references are not rendered as links. Sphinx gives some unrelated warnings, but does not complain about the references. It might be a configuration option. The sphinx warnings are related to the docstring indentation of some of the older estimators. I think the they could be fixed by using the latest version of `black` (which also considers docstrings to some degree) on them.

The full access path should still make it easier to find the relevant documentation.

timokau commented:
Thank you for the review :)
